internal/services/elastic/elastic_monitor_resource.go

```diff
@@ -0,0 +1,444 @@
+									},
+									"kibana_service_url": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"kibana_sso_url": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"sku": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+
+			"user_info": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"email_address": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validate.ElasticEmailAddress,
+						},
+					},
+				},
+			},
+
+			"monitoring_status": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  true,
+				ForceNew: true,
+			},
+
+			"liftr_resource_category": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"liftr_resource_preference": {
+				Type:     pluginsdk.TypeInt,
+				Computed: true,
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceElasticMonitorCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	client := meta.(*clients.Client).Elastic.MonitorClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	id := parse.NewElasticMonitorID(subscriptionId, resourceGroup, name)
+
+	existing, err := client.Get(ctx, id.ResourceGroup, id.MonitorName)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("checking for existing %q: %+v", id, err)
+		}
+	}
+	if !utils.ResponseWasNotFound(existing.Response) {
+		return tf.ImportAsExistsError("azurerm_elastic_monitor", id.ID())
+	}
+
+	monitoringStatus := elastic.MonitoringStatusDisabled
+	if d.Get("monitoring_status").(bool) {
+		monitoringStatus = elastic.MonitoringStatusEnabled
+	}
+
+	body := elastic.MonitorResource{
+		Location: utils.String(location.Normalize(d.Get("location").(string))),
+		Sku:      expandMonitorResourceSku(d.Get("sku").([]interface{})),
+		Properties: &elastic.MonitorProperties{
+			UserInfo:         expandMonitorUserInfo(d.Get("user_info").([]interface{})),
+			MonitoringStatus: monitoringStatus,
+		},
+		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+	future, err := client.Create(ctx, id.ResourceGroup, id.MonitorName, &body)
+	if err != nil {
+		return fmt.Errorf("creating Elastic Monitor %q (Resource Group %q): %+v", name, resourceGroup, err)
```

Comment:
This can be simplified to `return fmt.Errorf("creating %q: %+v", id, err)`

Reply:
Will be updated in the next commit